### PR TITLE
specs for wait for first consumer

### PR DESCRIPTION
**Pull request by Rohit-PX**
**What type of PR is this?**
integration-test

**What this PR does / why we need it**:
Verifying: https://github.com/kubevirt/containerized-data-importer/blob/main/doc/waitforfirstconsumer-storage-handling.md

https://portworx.atlassian.net/browse/PTX-20952

**Does this PR change a user-facing CRD or CLI?**:
no

**Is a release note needed?**:
no

**Does this change need to be cherry-picked to a release branch?**:
no

### CODE REVIEW

File: test/integration_test/specs/kubevirt-fedora-wait-first-consumer/fedora-vm-wait-first-consumer.yaml [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            #cloud-config
+            user: fedora
+            password: password1
+            chpasswd: { expire: False }
+        name: cloudinitdisk
+  dataVolumeTemplates:      
+  - metadata:
+      name: fedora-datavolume-original
+      annotations:
+        cdi.kubevirt.io/storage.bind.immediate.requested: "true"
```
Comment (pureneelesh) [inline test/integration_test/specs/kubevirt-fedora-wait-first-consumer/fedora-vm-wait-first-consumer.yaml:66]: This overrides WFFC. Should remove it.

File: test/integration_test/specs/kubevirt-templates/fedora/fedora-pvc-template-wait-first-consumer.yaml [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  annotations:
+    cdi.kubevirt.io/storage.bind.immediate.requested: "true"
+    cdi.kubevirt.io/storage.import.endpoint: https://ftp-nyc.osuosl.org/pub/fedora/linux/releases/37/Cloud/x86_64/images/Fedora-Cloud-Base-37-1.7.x86_64.qcow2
+spec:
+  storageClassName: sc-sharedv4svc-nolock-wait-first-consumer
+  accessModes:
+  - ReadWriteMany
+  resources:
+    requests:
+      storage: 50Gi
```
Comment (pureneelesh) [inline test/integration_test/specs/kubevirt-templates/fedora/fedora-pvc-template-wait-first-consumer.yaml:17]: We don't need this file. Template PVC will never use WFFC. Here, you are overriding it with the annotation anyways.

File: test/integration_test/specs/kubevirt-fedora-wait-first-consumer/fedora-vm-wait-first-consumer.yaml [hunk 3]
```diff
@@ -0,0 +1,79 @@
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  labels:
+    app: fedora-vm
+    vm.kubevirt.io/template: fedora-server-small
+    vm.kubevirt.io/template.namespace: openshift
+  name: fedora-test-vm-wait-first-consumer
+  annotations:
+    cdi.kubevirt.io/storage.bind.immediate.requested: "true"
```
Comment (pureneelesh) [inline test/integration_test/specs/kubevirt-fedora-wait-first-consumer/fedora-vm-wait-first-consumer.yaml:10]: We should not need this annotation. Does it not work without this?

File: test/integration_test/specs/kubevirt-templates/fedora/fedora-pvc-template-wait-first-consumer.yaml (hunk 2 above)
Comment (Rohit-PX) [reply test/integration_test/specs/kubevirt-templates/fedora/fedora-pvc-template-wait-first-consumer.yaml]: Removed.

File: test/integration_test/specs/kubevirt-fedora-wait-first-consumer/fedora-vm-wait-first-consumer.yaml (hunk 3 above)
Comment (Rohit-PX) [reply test/integration_test/specs/kubevirt-fedora-wait-first-consumer/fedora-vm-wait-first-consumer.yaml]: Removed this annotation and it seems to work without it as well.

File: test/integration_test/specs/kubevirt-fedora-wait-first-consumer/fedora-vm-wait-first-consumer.yaml (hunk 1 above)
Comment (Rohit-PX) [reply test/integration_test/specs/kubevirt-fedora-wait-first-consumer/fedora-vm-wait-first-consumer.yaml]: Removed.

File: test/integration_test/kubevirt_test.go [hunk 4]
```diff
@@ -27,9 +27,13 @@ const (
 	importerPodCompletionTimeout = 20 * time.Minute
 	importerPodRetryInterval     = 10 * time.Second
 
+	kubevirtTemplates                     = "kubevirt-templates"
 	kubevirtTemplateNamespace             = "openshift-virtualization-os-images"
 	kubevirtCDIStorageConditionAnnotation = "cdi.kubevirt.io/storage.condition.running.reason"
 	kubevirtCDIStoragePodPhaseAnnotation  = "cdi.kubevirt.io/storage.pod.phase"
+
+	volumeBindingImmediate         = "kubevirt-templates"
+	volumeBindingWaitFirstConsumer = "kubevirt-templates-wait-first-consumer"
```
Comment (pureneelesh) [inline test/integration_test/kubevirt_test.go:36]: unused, remove.

**Review by pureneelesh — APPROVED** [review]

**Review by pp511 — APPROVED** [review]
lgtm

**Pull request merged**